librz/core/golang.c

```diff
@@ -515,12 +512,12 @@ static bool add_new_bin_string(RzCore *core, char *string, ut64 vaddr, ut32 size
 	}
 	bstr->paddr = paddr;
 	bstr->vaddr = vaddr;
-	bstr->ordinal = rz_list_length(bf->o->strings);
+	bstr->ordinal = rz_list_length(strings);
 	bstr->length = bstr->size = size;
 	bstr->string = string;
 	bstr->type = RZ_STRING_ENC_UTF8;
-	if (!rz_list_append(bf->o->strings, bstr)) {
-		RZ_LOG_ERROR("Failed append new go string to strings list\n");
+	if (!rz_bin_object_string_add(bin, bf->o, bstr)) {
```

Comment:
I think it is questionable whether this kind of adding strings to a binobject should be possible at all, since the info in it is expected to be statically derived from the bin, not through analysis in core. But I don't know a definitive solution yet either so I am ok with keeping the adding in this pr.
But since this is currently only done in this single place and to avoid future code like this, I suggest to remove rz_bin_object_string_add() and call `rz_bin_string_database_add()` directly on the contained db.